python/lsst/analysis/tools/atools/cpVerifyQuantityProfile.py

```diff
@@ -0,0 +1,155 @@
+    )
+    dataKey = Field[str](
+        doc="Data selector. Data will be separated into multiple groups in a single panel based on this key.",
+    )
+    quantityKey = Field[str](
+        doc="Quantity selector. The actual data quantities to be plotted.",
+    )
+
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        repackedData = {}
+        # Loop over the length of the data vector and repack row by row
+        for i in range(len(cast(Vector, data[self.panelKey]))):
+            panelVec = cast(Vector, data[self.panelKey])
+            dataVec = cast(Vector, data[self.dataKey])
+            quantityVec = cast(Vector, data[self.quantityKey])
+            repackedData[f"{panelVec[i]}_{dataVec[i]}_{self.quantityKey}"] = quantityVec[i]
+        return repackedData
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return (
+            (self.panelKey, Vector),
+            (self.dataKey, Vector),
+            (self.quantityKey, Vector),
+        )
+
+    def addInputSchema(self, inputSchema: KeyedDataSchema) -> None:
+        pass
+
+
+class PassThrough(KeyedDataAction):
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        return data
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        # In general this method should be implemented, but here we are ALSO
+        # implementing the prep step. We therefore know the method results are
+        # not needed because it is doing something special with the inputs.
+        return ()
+
+
+class CPVerifyQuantityBaseTool(AnalysisTool):
+    parameterizedBand: bool = False
+
+    plotElement = ConfigurableActionField[PlotElement](
+        doc="Plot element.",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Repack the input data into a usable format
+        self.prep = PrepRepacker()
+
+        # A simple pass-through process action to keep the data unchanged
+        self.process = PassThrough()
+
+        # Plot the repacked data in a 4x4 grid
+        self.produce.plot = GridPlot()
+        self.produce.plot.plotElements = {}
+        self.produce.plot.numRows = 4
+        self.produce.plot.numCols = 4
+
+        # Values to group by to distinguish between data in differing panels
+        self.produce.plot.valsGroupBy = {
+            0: "C00",
+            1: "C01",
+            2: "C02",
+            3: "C03",
+            4: "C04",
+            5: "C05",
+            6: "C06",
+            7: "C07",
+            8: "C10",
+            9: "C11",
+            10: "C12",
+            11: "C13",
+            12: "C14",
+            13: "C15",
+            14: "C16",
+            15: "C17",
+        }
+
+    def finalize(self):
+        super().finalize()
+
+        # Set the plot element for each panel to an AxPlotElement
+        for key, value in self.produce.plot.valsGroupBy.items():
+            plotElementConfig = PlotElementConfig(
+                plotElement=self.plotElement,
```

Comment:
Am I correct that if I subclassed this, and set `plotElement` based on some arbitrary method, I could mix the `ScatterElement` and `HistElement` in one `GridPlot`?

Reply:
Yes, that's absolutely correct. This is indeed something @natelust and I played around with during testing of `GridPlot`. However, you're also not forced to use `GridPlot` at all - you could equally define your own plot action that calls exactly the plot elements you need for the plot you have in mind.

Reply:
PS - the `plotElementConfig` has been renamed to `gridPanelConfig`, and the grid plot `plotElements` argument has been renamed to `panels`. The over-use of `plotElement` was beginning to get a bit confusing for me, so I thought to broaden our namespace now to more appropriate and descriptive names.

Reply:
The new names make sense.  I usually try not to review names unless they're completely confusing, but this seems like a nice improvement.